pymess/backend/sms/ats_sms_operator.py

```diff
@@ -230,30 +230,32 @@ def _update_sms_states_from_response(self, messages, parsed_response, is_sending
                 )
 
     def publish_messages(self, messages):
+        messages = (message for message in messages if self._is_recipient_phone_number_allowed(message))
         self._send_requests(messages, request_type=self.REQUEST_TYPES.SMS, is_sending=True, sent_at=timezone.now())
 
     def publish_message(self, message):
-        try:
-            self._send_requests(
-                [message],
-                request_type=self.REQUEST_TYPES.SMS,
-                is_sending=True,
-                sent_at=timezone.now()
-            )
-        except self.ATSSendingError as ex:
-            self._update_message_after_sending_error(
-                message,
-                state=OutputSMSMessage.STATE.ERROR,
-                error=str(ex),
-            )
-        except requests.exceptions.RequestException as ex:
-            # Service is probably unavailable sending will be retried
-            self._update_message_after_sending_error(
-                message,
-                error=str(ex)
-            )
-            # Do not re-raise caught exception. Re-raise exception causes transaction rollback (lost of information
-            # about exception).
+        if self._is_recipient_phone_number_allowed(message):
```

Comment:
I think you should do this check in parent, otherwise in new backends, it will be easy to forget calling this. Something like:
```
def publish_message(self, message):
    if self._can_publish_message(message):
        self._publish_message(message)

def _publish_message(self, message):
    raise NotImplementedError
```

Reply:
better will be solution with dynamic backends

Reply:
umm, ok, but that will take some time to rewrite